toolchain/check/import_ref.cpp

```diff
@@ -125,44 +125,58 @@ auto VerifySameCanonicalImportIRInst(Context& context, SemIR::InstId prev_id,
 // Calling Resolve on an instruction operates in an iterative manner, tracking
 // Work items on work_stack_. At a high level, the loop is:
 //
-// 1. If a constant value is already known for the work item, and we're
-//    processing it for the first time, it's considered resolved.
+// 1. If a constant value is already known for the work item and was not set by
+//    this work item, it's considered resolved.
 //    - The constant check avoids performance costs of deduplication on add.
-//    - If `retry` is set, we process it again, because it didn't complete last
-//      time, even though we have a constant value already.
-// 2. Resolve the instruction: (TryResolveInst/TryResolveTypedInst)
-//    - For a symbolic constant within a generic, find the generic itself. If it
-//      needs to be imported, return Retry() to import the generic before we
-//      import the constant.
-//    - For instructions that can be forward declared, if we don't already have
-//      a constant value from a previous attempt at resolution, start by making
-//      a forward declared constant value to address circular references.
-//    - Gather all input constants.
+//    - If we've processed this work item before, then we now process it again,
+//      because it didn't complete last time, even though we have a constant
+//      value already.
+//
+// 2. Resolve the instruction (TryResolveInst/TryResolveTypedInst), in up to
+//    three passes. First pass:
```

Comment:
Had you considered structuring this more like:

```
- First pass:
  - ...
  - ...
- Second pass:
  - ...
```

Reply:
Done.